clients/js/test/plugins/collection/updateDelegate.test.ts

```diff
@@ -337,3 +540,167 @@ test('it updateDelegate on collection cannot update an asset after delegate auth
   });
 });
+
+test('it can update a non-updateDelegate plugin on an asset as collection update additional delegate', async (t) => {
+  const umi = await createUmi();
+  const updateDelegate = await generateSignerWithSol(umi);
+
+  const { asset, collection } = await createAssetWithCollection(
+    umi,
+    {
+      plugins: [
+        {
+          type: 'Edition',
+          number: 1,
+        },
+      ],
+    },
+    {
+      plugins: [
+        {
+          type: 'UpdateDelegate',
+          additionalDelegates: [updateDelegate.publicKey],
+        },
+      ],
+    }
+  );
+
+  await updatePlugin(umi, {
+    asset: asset.publicKey,
+    collection: collection.publicKey,
+    plugin: {
+      type: 'Edition',
+      number: 2,
+    },
+    authority: updateDelegate,
+  }).sendAndConfirm(umi);
+
+  await assertAsset(t, umi, {
+    ...DEFAULT_ASSET,
+    asset: asset.publicKey,
+    owner: umi.identity.publicKey,
+    updateAuthority: { type: 'Collection', address: collection.publicKey },
+    edition: {
+      authority: { type: 'UpdateAuthority' },
+      number: 2,
+    },
+  });
+});
+
+test('it can approve/revoke non-updateDelegate plugin on an asset as collection update additional delegate', async (t) => {
+  const umi = await createUmi();
+  const updateDelegate = await generateSignerWithSol(umi);
+  const updateDelegate2 = generateSigner(umi);
+
+  const { asset, collection } = await createAssetWithCollection(
+    umi,
+    {
+      plugins: [
+        {
+          type: 'Edition',
+          number: 1,
+        },
+      ],
+    },
+    {
+      plugins: [
+        {
+          type: 'UpdateDelegate',
+          additionalDelegates: [updateDelegate.publicKey],
+        },
+      ],
+    }
+  );
+
+  await approvePluginAuthority(umi, {
+    asset: asset.publicKey,
+    collection: collection.publicKey,
+    plugin: {
+      type: 'Edition',
+    },
+    newAuthority: { type: 'Address', address: updateDelegate2.publicKey },
+    authority: updateDelegate,
+  }).sendAndConfirm(umi);
+
+  await assertAsset(t, umi, {
+    ...DEFAULT_ASSET,
+    asset: asset.publicKey,
+    owner: umi.identity.publicKey,
+    updateAuthority: { type: 'Collection', address: collection.publicKey },
+    edition: {
+      authority: { type: 'Address', address: updateDelegate2.publicKey },
+      number: 1,
+    },
+  });
+
+  await revokePluginAuthority(umi, {
+    asset: asset.publicKey,
+    collection: collection.publicKey,
+    plugin: {
+      type: 'Edition',
+    },
+    authority: updateDelegate,
+  }).sendAndConfirm(umi);
+
+  await assertAsset(t, umi, {
+    ...DEFAULT_ASSET,
+    asset: asset.publicKey,
+    owner: umi.identity.publicKey,
+    updateAuthority: { type: 'Collection', address: collection.publicKey },
+    edition: {
+      authority: { type: 'UpdateAuthority' },
+      number: 1,
+    },
+  });
+});
+
+test('it cannot update the update authority of the collection as an updateDelegate additional delegate', async (t) => {
```

Comment:
Should there be an Asset version of this test where it cannot update the update authority of the Asset (when authority type is Address {...}?

Reply:
Yeah that makes sense.

Reply:
Yes aaaaand it failed so fixing. Good catch.

Reply:
I see 'it cannot update the update authority of the asset as an updateDelegate additional delegate' has been added.  Recommend adding one called 'it cannot update the update authority of the asset as an updateDelegate' that doesn't let the updateDelegate root authority do it either.

Reply:
Good idea, added!